internal/cmd/connector_test.go

```diff
@@ -0,0 +1,269 @@
+			{User: "user1@example.com", Resource: "testing.ns1", Role: "admin"},
+			{User: "user2@example.com", Resource: "testing", Role: "view"},
+			{Group: "group1@example.com", Resource: "testing.ns1", Role: "logs"},
+		},
+	}
+
+	srv, err := server.New(serverOpts)
+	assert.NilError(t, err)
+
+	fakeKube := &fakeKubeAPI{t: t}
+	kubeSrv := httptest.NewTLSServer(fakeKube)
+	t.Cleanup(kubeSrv.Close)
+
+	ctx := context.Background()
+	runAndWait(ctx, t, srv.Run)
+
+	opts := connector.Options{
+		Server: connector.ServerOptions{
+			URL:                srv.Addrs.HTTPS.String(),
+			AccessKey:          "0000000002.connectorconnectorconnec",
+			TrustedCertificate: serverOpts.TLS.Certificate,
+		},
+		Name:         "testing",
+		CACert:       types.StringOrFile(readFile(t, "testdata/pki/connector.crt")),
+		CAKey:        types.StringOrFile(readFile(t, "testdata/pki/connector.key")),
+		EndpointAddr: types.HostPort{Host: "127.0.0.1", Port: 55555},
+		Kubernetes: connector.KubernetesOptions{
+			AuthToken: "auth-token",
+			Addr:      kubeSrv.URL,
+			CA:        types.StringOrFile(certs.PEMEncodeCertificate(kubeSrv.Certificate().Raw)),
+		},
+		Addr: connector.ListenerOptions{
+			HTTPS:   "127.0.0.1:0",
+			Metrics: "127.0.0.1:0",
+		},
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	t.Cleanup(cancel)
+
+	runAndWait(ctx, t, func(ctx context.Context) error {
+		return connector.Run(ctx, opts)
+	})
+
+	// check destination has been registered
+	var destination *models.Destination
+	poll.WaitOn(t, func(t poll.LogT) poll.Result {
+		destination, err = data.GetDestination(srv.DB(), data.ByName("testing"))
+		switch {
+		case errors.Is(err, internal.ErrNotFound):
+			return poll.Continue("destination not registered")
+		case err != nil:
+			return poll.Error(err)
+		}
+		return poll.Success()
+	})
+
+	// check the destination was updated
+	expected := &models.Destination{
+		Model: models.Model{
+			ID:        anyUID,
+			CreatedAt: time.Now(),
+			UpdatedAt: time.Now(),
+		},
+		OrganizationMember: models.OrganizationMember{OrganizationID: srv.DB().OrganizationID()},
+		Name:               "testing",
+		UniqueID:           "4ebfd7dabeec5b37eafd20e3775f70ab86c7422036367d77d9bebfa03864e08b",
+		ConnectionURL:      "127.0.0.1:55555",
+		ConnectionCA:       opts.CACert.String(),
+		LastSeenAt:         time.Now(),
+		Version:            "99.99.99999",
+		Resources:          models.CommaSeparatedStrings{"default", "ns1", "ns2"},
+		Roles:              models.CommaSeparatedStrings{"admin", "view", "edit", "custom", "logs"},
+	}
+	assert.DeepEqual(t, destination, expected, cmpDestinationModel)
+
+	poll.WaitOn(t, func(t poll.LogT) poll.Result {
+		fakeKube.writesLock.Lock()
+		defer fakeKube.writesLock.Unlock()
+		if len(fakeKube.writes) >= 3 {
+			return poll.Success()
+		}
+		return poll.Continue("request count %d waiting for 3", len(fakeKube.writes))
+	})
+
+	// check kube bindings were updated
+	expectedWrites := []*http.Request{
```

Comment:
No assertions on the subject?

Reply:
Makes sense, I guess that's in the body of the request?